src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractTranslator.java

```diff
@@ -342,6 +343,34 @@ public String translate(Update u) {
         return join(temp, " ");
     }
 
+    /**
+     * Translates {@link UpdateFrom}.
+     *
+     * @param updateFrom The object to translate.
+     * @return The string representation of the given object.
+     */
+    public String translate(final UpdateFrom updateFrom) {
+        final Expression from = updateFrom.getFrom();
+
+        if (from == null) {
+            return translate((Update) updateFrom);
+        }
+
+        inject(from);
+
+        final String translate = translate((Update) updateFrom);
+        final StringBuilder stringBuilder = new StringBuilder(translate);
+
+        final int where = translate.indexOf("WHERE");
+        if (where == -1) {
```

Comment:
instead of doing this weird thing, and having a new "UpdateFrom" subclass of `Update`,
why not just add a "from" method in that class, and handle this properly in the translation for Update?

Reply:
As you noticed, only PostgreSQL and DB2 has a "clean" way of doing this. So I don't think we should pollute the regular `Update` with the odd way we can do it in the other engines, hence the `UpdateFrom`